heudiconv/heuristics/convertall.py

```diff
@@ -1,12 +1,29 @@
+import logging
 import os
 
+lgr = logging.getLogger('heudiconv')
+
 
 def create_key(template, outtype=('nii.gz',), annotation_classes=None):
     if template is None or not template:
         raise ValueError('Template must be a valid format string')
     return template, outtype, annotation_classes
 
 
+def custom_seqinfo(series_files, wrapper, **kw):
+    # Just a dummy demo for what custom_seqinfo could get/do
+    # for already loaded DICOM data, and including storing/returning
+    # the sample series file as was requested
+    # in https://github.com/nipy/heudiconv/pull/333
+    from nibabel.nicom.dicomwrappers import WrapperError
+    try:
+        affine = wrapper.affine.tostring()
+    except WrapperError:
+        lgr.exception("Errored out while obtaining/converting affine")
+        affine = None
+    return affine, series_files[0]
```

Comment:
sneaky sneaky @bpinsard !! ;) this might go into docs on `custom_seqinfo` but should not become part of `convertall` heuristic IMHO! ;-)

may be middle ground (and useful for another demonstration) could be creating a `converall-custom.py` heuristic which would be

```python
"""A demo convertall heuristic with custom_seqinfo extracting affine and sample DICOM path

This heuristic also demonstrates on how to create a "derived" heuristic which would augment
behavior of an already existing heuristic without complete rewrite.  Such approach could be
useful for heuristic like  reproin  to overload mapping etc.
"""

from .convertall import *

def custom_seqinfo(series_files, wrapper, **kw):
    # Just a dummy demo for what custom_seqinfo could get/do
    # for already loaded DICOM data, and including storing/returning
    # the sample series file as was requested
    # in https://github.com/nipy/heudiconv/pull/333
    from nibabel.nicom.dicomwrappers import WrapperError
    try:
        affine = wrapper.affine.tostring()
    except WrapperError:
        lgr.exception("Errored out while obtaining/converting affine")
        affine = None
    return affine, series_files[0]
```
?

Reply:
@bpinsard ping on above

Reply:
Very sorry for the late follow-up.

I am not sure to understand what you are asking me.
In that commit https://github.com/nipy/heudiconv/pull/581/commits/f104fbcda275b2b8e336ba3cf1b424bb9fb8cef0 I  tried fixing that function which you added to the convertall heuristics in the PR, as the tests were crashing on the test data (due to a missing affine in the header). If we want the tests to pass it seems necessary. Or maybe I misunderstand your point here.

Reply:
ha ha -- it indeed was me who added it in 3bc1f4e58ef574dcf6565b6c47a74be6c7eeab46 but I did mark it as TMP (i.e dropped for final version)

```
commit 3bc1f4e58ef574dcf6565b6c47a74be6c7eeab46
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Tue Jul 19 16:10:25 2022 -0400

    TMP: just a demonstration on how custom_seqinfo could/should be used
```

now pushed as that proposed new heuristic